packages/app/src/lib/import.ts

```diff
@@ -240,13 +251,118 @@ export async function asLastPass(data: string): Promise<VaultItem[]> {
+        return;
+    }
+
+    const itemName = rowData.name;
+    const tags = rowData.tags.split(",");
+
+    if (item.trashed) {
+        tags.push("trashed");
+    }
+
+    let fields: Field[] = [
+        new Field({ name: $l("Username"), value: rowData.username, type: FieldType.Username }),
+        new Field({ name: $l("Password"), value: rowData.password, type: FieldType.Password }),
+        new Field({ name: $l("URL"), value: rowData.url, type: FieldType.Url }),
+    ];
+
+    if (rowData.notes) {
+        fields.push(new Field({ name: $l("Notes"), value: rowData.notes, type: FieldType.Note }));
+    }
+
+    for (const extraField of rowData.extraFields) {
+        if (extraField.type === "totp") {
+            // Extract just the secret
+            try {
+                const secret = new URL(extraField.value).searchParams.get("secret");
+                if (secret) {
+                    fields.push(new Field({ name: extraField.name, value: secret, type: FieldType.Totp }));
+                }
+            } catch (error) {
+                // Do nothing
+            }
+        } else {
+            fields.push(
+                new Field({ name: extraField.name, value: extraField.value, type: extraField.type as FieldType })
+            );
+        }
+    }
+
+    return createVaultItem(itemName, fields, tags);
+}
+
+export async function as1Pux(file: File): Promise<VaultItem[]> {
+    try {
+        const { parse1PuxFile } = await import("./1pux-parser");
+        const data = await readFileAsArrayBuffer(file);
+        const dataExport = await parse1PuxFile(data);
+
+        const items = [];
+
+        for (const account of dataExport.data.accounts) {
+            for (const vault of account.vaults) {
+                for (const vaultItem of vault.items) {
+                    if (vaultItem.item) {
+                        const parsedItem = await parse1PuxItem(account.attrs.name, vault.attrs.name, vaultItem.item);
+                        if (parsedItem) {
+                            items.push(parsedItem);
+                        }
+                    }
+                }
+            }
+        }
+
+        return items;
+    } catch (error) {
+        throw new Err(ErrorCode.INVALID_1PUX, "Failed to parse .1pux file.");
+    }
+}
+
 /**
- * Checks if a given string represents a LastPass CSV file
+ * Checks if a given file name ends with .1pux to avoid trying to parse unnecessarily
  */
-export function isLastPass(data: string): boolean {
-    return data.split("\n")[0] === "url,username,password,extra,name,grouping,fav";
+export function is1Pux(file: File): boolean {
+    return file.name.endsWith(".1pux");
 }
 
-export function guessFormat(data: string): ImportFormat | null {
-    return isPBES2Container(data) ? PBES2 : isPadlockV1(data) ? PADLOCK_LEGACY : isLastPass(data) ? LASTPASS : CSV;
+export async function guessFormat(file: File): Promise<ImportFormat> {
+    // Try to guess sync first (won't need parsing)
```

Comment:
So, one problem of moving to `file` being passed down and delaying the reading of files is that for ones that can't be distinguished from file extensions (like lastpass/csv), the parsing needs to happen multiple times, if it keeps failing to guess the format. It should not be slow or a big problem, though.